client/js/_author/components/assessments/question_types/drag_and_drop/drop_zone.jsx

```diff
@@ -0,0 +1,245 @@
+import React      from 'react';
+import _          from 'lodash';
+
+export default class DropZone extends React.Component {
+  static propTypes = {
+    zone: React.PropTypes.shape({
+      id: React.PropTypes.string,
+    }).isRequired,
+    target: React.PropTypes.shape({
+      getBoundingClientRect: React.PropTypes.func,
+    }),
+    editZone: React.PropTypes.func.isRequired,
+    setActive: React.PropTypes.func.isRequired,
+    isActive: React.PropTypes.bool,
+  };
+
+  static saveZoneToState(zone) {
+    return {
+      leftPos: zone.xPos,
+      topPos: zone.yPos,
+      rightPos: zone.xPos + zone.width,
+      bottomPos: zone.yPos + zone.height,
+      initialX: null,
+      initialY: null,
+    };
+  }
+
+  static boundaryCheck(pos, boundary) {
+    if (pos < 0) { return 0; }
+    if (pos > boundary) { return boundary; }
+    return pos;
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = DropZone.saveZoneToState(props.zone);
+  }
+
+  zonePosition() {
+    const { leftPos, topPos, rightPos, bottomPos } = this.state;
```

Comment:
That makes the functions pretty long, and I have to newline them. Destructuring state is cleaner imho, but ok.